src/nunit.analyzers.codefixes/UseAssertEnterMultipleScope/UseAssertEnterMultipleScopeCodeFix.cs

```diff
@@ -0,0 +1,159 @@
+            if (expressionStatementSyntax is null)
+            {
+                return;
+            }
+
+            var (blockSyntax, lambdaHasAsyncKeyword) = FindBlockSyntax(node);
+            if (blockSyntax is null)
+            {
+                return;
+            }
+
+            var newBodySyntax = CreateUsingStatementSyntax(blockSyntax, expressionStatementSyntax.GetTrailingTrivia());
+
+            // Replace body syntax and add annotation to have a reference to the new body syntax in the updated root
+            var annotation = new SyntaxAnnotation();
+            var annotatedNewSyntax = newBodySyntax.WithAdditionalAnnotations(annotation);
+            var rootWithAnnotatedBody = root.ReplaceNode(expressionStatementSyntax, annotatedNewSyntax);
+            var newBodySyntaxInUpdatedRoot = rootWithAnnotatedBody.GetAnnotatedNodes(annotation).First();
+
+            var newRoot = UpdateTestMethodSignatureIfNecessary(rootWithAnnotatedBody, newBodySyntaxInUpdatedRoot, lambdaHasAsyncKeyword);
+
+            var codeAction = CodeAction.Create(
+                UseAssertEnterMultipleScopeMethod,
+                _ => Task.FromResult(context.Document.WithSyntaxRoot(newRoot)),
+                UseAssertEnterMultipleScopeMethod);
+
+            context.RegisterCodeFix(codeAction, context.Diagnostics);
+        }
+
+        private static (BlockSyntax? Block, bool HasAsyncKeyword) FindBlockSyntax(SyntaxNode node)
+        {
+            if (node is not InvocationExpressionSyntax invocationExprSyntax ||
+                invocationExprSyntax.ArgumentList.Arguments.Count != 1 ||
+                invocationExprSyntax.ArgumentList.Arguments[0].Expression
+                    is not ParenthesizedLambdaExpressionSyntax parenthesizedLambdaExpressionSyntax)
+            {
+                return (null, false);
+            }
+
+            return (parenthesizedLambdaExpressionSyntax.Block, parenthesizedLambdaExpressionSyntax.AsyncKeyword != default);
+        }
+
+        private static UsingStatementSyntax CreateUsingStatementSyntax(BlockSyntax blockSyntax, SyntaxTriviaList trailingTrivia) =>
+            SyntaxFactory.UsingStatement(blockSyntax)
+                .WithExpression(
+                    SyntaxFactory.InvocationExpression(
+                        SyntaxFactory.MemberAccessExpression(
+                            SyntaxKind.SimpleMemberAccessExpression,
+                            SyntaxFactory.IdentifierName(NUnitFrameworkConstants.NameOfAssert),
+                            SyntaxFactory.IdentifierName(NUnitV4FrameworkConstants.NameOfEnterMultipleScope))))
+                .WithTrailingTrivia(trailingTrivia);
+
+        private static SyntaxNode UpdateTestMethodSignatureIfNecessary(
+            SyntaxNode newRoot,
+            SyntaxNode newSyntaxInTree,
+            bool lambdaHasAsyncKeyword)
+        {
+            var methodDeclaration = FindNearestParentOfType<MethodDeclarationSyntax>(newSyntaxInTree);
```

Comment:
Use existing method:
```suggestion
            var methodDeclaration = newSyntaxInTree.FirstAncestorOrSelf<MethodDeclarationSyntax>();
(newSyntaxInTree);
```